### ISSUE TITLE
[HOLD component refactoring] [$500] [LOW] Green circular loading does not appear when searching for a contact in request money flow

### ISSUE THREAD

**Issue by m-natarajan** [Expensify/App#37144]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 1.4.43-14
**Reproducible in staging?:** y
**Reproducible in production?:** y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:** @brunovjk 
**Slack conversation:** https://expensify.slack.com/archives/C01GTK53T8Q/p1708081508978019

## Action Performed:
1. Click +  and select `Request money`
2. Enter amount and click next
3. Search for any contact in the field

## Expected Result:
Green circular loading ring is seen like when we search for a contact to chat

## Actual Result:
Not appearing when searching in `Request money` flow

## Workaround:
unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [ ] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence

https://github.com/Expensify/App/assets/38435837/69a0d4f2-5176-4a87-bcd4-9aaacae07549

</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01788bcdaa345e5022</li>
        <li>Upwork Job ID: 1761057209378701312</li>
        <li>Last Price Increase: 2024-02-23</li>
    </ul>
</details>

**Comment by FitseTLT** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Green circular loading does not appear when searching for a contact in request money flow

### What is the root cause of that problem?
We are not passing `isLoadingNewOptions` prop to the selection list
### What changes do you think we should make in order to solve the problem?

We should be passing `isLoadingNewOptions` prop to the selection list as for search contact
### What alternative solutions did you explore? (Optional)

**Comment by brunovjk** [comment]
I don't want to cause any problems here and I accept the review's decision. But I would really like to work on this issue, I noticed this bug a month ago, but I was waiting for [Selection List Refactor](https://github.com/Expensify/App/issues/20354).
[Report issue on slack](https://expensify.slack.com/archives/C01GTK53T8Q/p1708081508978019).
Thanks in advance

**Comment by brunovjk** [comment]
I don't want to cause any problems here and I accept the review's decision. But I would really like to work on this issue, I noticed this bug a month ago, but I was waiting for [Selection List Refactor](https://github.com/Expensify/App/issues/20354) so we can fix it.
[Report issue on slack](https://expensify.slack.com/archives/C01GTK53T8Q/p1708081508978019).
Thanks in advance

**Comment by mollfpr** [comment]
Just a head-ups! I'll be reviewing the proposal from @brunovjk first since the posted the proposal along side with this bug report. Context: https://expensify.slack.com/archives/C01GTK53T8Q/p1708081508978019

**Comment by jliexpensify** [comment]
Posted to the Wave6 channel - https://expensify.slack.com/archives/C02MW39LT9N/p1708908540463509

**Comment by mollfpr** [comment]
Thank you guys for your patience!

The [proposal](https://github.com/Expensify/App/issues/37144#issuecomment-1961608043) from @brunovjk looks good to me! I'm also inclined to update `MoneyRequestParticipantsSelector` to have `isLoadingNewOptions`, so we can get the same behavior for the send money flow.

🎀 👀 🎀 C+ reviewed!

**Comment by mollfpr** [comment]
My consideration is the main issue here is the `MoneyTemporaryForRefactorRequestParticipantsSelector` missing `isLoadingNewOptions`. So basically the proposal from @brunovjk has solved the issue, and the other proposal seems doing the same.

Yes, your proposal has an addition to fix the send money flow, but because it's not the main issue, the fix is pretty similar. So I don't think it's fair for another contributor that focus only on the main problem here.

I let @lakchote decide who's gonna work on this issue.

**Comment by brunovjk** [comment]
@mollfpr Thanks for the review.

> In the Send money flow [here](https://github.com/Expensify/App/blob/c71387c6dc3d2baf052d02701db650cebc397d86/src/pages/iou/steps/MoneyRequstParticipantsPage/MoneyRequestParticipantsSelector.js#L361), we have the Skeleton loading whenever the user types something to search

@dukenv0307 I believe that is not the scope of this issue. I dont think we are oing to use `MoneyRequestParticipantsSelector`, instead we will use `MoneyTemporaryForRefactorRequestParticipantsSelector`. If I'm not mistaken, we are still refactoring `SendMoney` flow :D

**Comment by brunovjk** [comment]
@mollfpr Thanks for the review.

> In the Send money flow [here](https://github.com/Expensify/App/blob/c71387c6dc3d2baf052d02701db650cebc397d86/src/pages/iou/steps/MoneyRequstParticipantsPage/MoneyRequestParticipantsSelector.js#L361), we have the Skeleton loading whenever the user types something to search

@mollfpr I didn't mention this because I thought it was out of scope of this issue. I dont think we are going to use `MoneyRequestParticipantsSelector`, instead we will use `MoneyTemporaryForRefactorRequestParticipantsSelector`. If I'm not mistaken, we are still refactoring `SendMoney` flow :D

**Comment by brunovjk** [comment]
@mollfpr Thanks for the review.

> In the Send money flow [here](https://github.com/Expensify/App/blob/c71387c6dc3d2baf052d02701db650cebc397d86/src/pages/iou/steps/MoneyRequstParticipantsPage/MoneyRequestParticipantsSelector.js#L361), we have the Skeleton loading whenever the user types something to search

@mollfpr I didn't mention this because I thought it was out of scope of this issue. I dont think we are going to use `MoneyRequestParticipantsSelector`, instead we will use `MoneyTemporaryForRefactorRequestParticipantsSelector`. If I'm not mistaken, we are still refactoring `Selection List SendMoney` flow :D

**Comment by mollfpr** [comment]
DM'ed @lakchote for the review!

**Comment by lakchote** [comment]
@brunovjk's proposal LGTM.

Thanks @mollfpr for the C+ review.

**Comment by brunovjk** [comment]
Thank you all for the review. PR will be ready for review within 2 days.

**Comment by brunovjk** [comment]
[PR](https://github.com/Expensify/App/pull/37842) ready for review :D

**Comment by brunovjk** [comment]
@dukenv0307 The component you are trying to change will be removed [here](https://github.com/Expensify/App/issues/34617).

> We won't remember/keep track of this to do it

Don't worry, I'll keep close eye on it, we won't forget :D

**Comment by brunovjk** [comment]
@dukenv0307 The component you are trying to change will be removed [here](https://github.com/Expensify/App/issues/34617).

**Comment by brunovjk** [comment]
@dukenv0307 The component you are trying to change will be removed [here](https://github.com/Expensify/App/issues/34617), and after _"Replace all uses of the Old Component with the New Component"_ the issue you are reporting will not exist.

**Comment by lakchote** [comment]
> @dukenv0307 The component you are trying to change will be removed [here](https://github.com/Expensify/App/issues/34617), and after _"Replace all uses of the Old Component with the New Component"_ the issue you are reporting will not exist.

How will it solve it? I do not agree. In the new component, there's the old logic in place, see [here](https://github.com/Expensify/App/blob/86a990428df2c923ed4ebc291c0a3a73653cb7fe/src/pages/iou/request/MoneyTemporaryForRefactorRequestParticipantsSelector.js#L349). 

Since no changes have been made to the logic in the old component, it won't be ported to the new component. 
So the issue will remain.

If we just solve the current issue and not the related one for the skeleton, we _risk_  letting a known problem be unfixed whereas we could act on it. That's not ideal actually. We want to have the same behavior across components. Especially related ones (Request/Send money components). The "other" issue reported by @dukenv0307 was found while he was investigating the bug. It's not like the two issues are entirely unrelated.

I'm going to seek for an external opinion on this. Thanks @dukenv0307 for pinging me again on this and raise my awareness of the problem.

**Comment by brunovjk** [comment]
> We want to have the same behavior across components

I can guarantee that, I just didn't think it was necessary for this issue, Im so sorry about that.

**Comment by mountiny** [comment]
@lakchote @brunovjk Given that this issue would prevail in the new component, I think we should put this issue on hold for the other one and then we can resume work once new component will be in place

**Comment by brunovjk** [comment]
Hey @lakchote, @mountiny and @dukenv0307,

Appreciate you delving into this with me. 🚀 Still getting the ropes here, and your insights are a huge help.

> How will it solve it? I do not agree. In the new component, there's the old logic in place, see [here](https://github.com/Expensify/App/blob/86a990428df2c923ed4ebc291c0a3a73653cb7fe/src/pages/iou/request/MoneyTemporaryForRefactorRequestParticipantsSelector.js#L349).

The line you referenced is from our [PR](https://github.com/Expensify/App/pull/32665). Also, I didn't move the new change to the old component based on @ntdyary's suggestion [link to the comment](https://github.com/Expensify/App/pull/32665#pullrequestreview-1827145169).

Please, correct me if Im wrong. In my view, we've got two distinct issues: green circular loading (our current discussion) and skeleton loading control (highlighted by @dukenv0307). Our issue involves the variable `isSearchingForReports` controlling `isLoadingNewOptions` for green circular loading. @dukenv0307issue deals with `didScreenTransitionEnd` and `isOptionsDataReady` controlling `showLoadingPlaceholder` for skeleton loading. The former triggers with every new contact search, while the latter activates only once during screen transition/list loading.

Currently, we have several flows loading contact lists using `IOURequestStepParticipants` (new component) and `MoneyRequestParticipantsPage` (old component). The transition/skeleton error was reported [here](https://github.com/Expensify/App/issues/31779), and during the PR, it surfaced in both components, as indicated by @ntdyary ([link to the comment](https://github.com/Expensify/App/pull/32665#pullrequestreview-1827145169)). However, he suggested not altering the old component as it's slated for [removal](https://github.com/Expensify/App/issues/29107).

In summary:

1. The skeleton loading issue is addressed in my latest [PR](https://github.com/Expensify/App/pull/32665), focusing on `MoneyTemporaryForRefactorRequestParticipantsSelector`. I see now that we should have dealt with this at once. Ensuring consistency between `MoneyRequestParticipantsSelector` and `MoneyTemporaryForRefactorRequestParticipantsSelector` might come with component refactoring, but your guidance suggest otherwise.
2. The green circular loading issue could be a separate task (I can update my PR to modify `MoneyRequestParticipantsSelector` as well).

I dont mind about splitting compensation. I just believe it'd be more effective to compartmentalize these into distinct purposes/issues.

**Comment by brunovjk** [comment]
But I also agree that we can _"put this issue on hold for the other one and then we can resume work once new component will be in place"_

**Comment by brunovjk** [comment]
Still waiting [Refactor navigation](https://github.com/Expensify/App/issues/29107)

**Comment by brunovjk** [comment]
Pages that use `SelectionList` to handle a options list | Skeleton | Green Circular |
| --- | --- | --- |
|  - Search page | <details> <summary>  ✅   </summary> <video src='https://github.com/Expensify/App/assets/95647348/c828cd43-324c-4ace-b885-2a060fb8bced'>  </details> | <details> <summary> ✅  </summary> <video src='https://github.com/Expensify/App/assets/95647348/4343e8b2-0a2b-4d4d-9863-17752a397928'>  </details> | 
|  - New chat | <details> <summary> ❌ should be fixed [here](https://github.com/Expensify/App/pull/38610) </summary> <video src='https://github.com/Expensify/App/assets/95647348/bafc369a-08d1-4480-be21-c728b7a3da65'> </details>| <details> <summary> ✅  </summary> <video src='https://github.com/Expensify/App/assets/95647348/5eb6585c-767c-4cb9-afc2-fef71952fc8d'> </details> |
| <details> <summary> New room chat - RoomInvitePage </summary> - Press "Start chat" </br> - Select "Room" </br> - Fill the name and create a room </br> - Open room details </br> - Press "Members" </br> - Press "Invite" </br> </details>  | <details> <summary>  ❌ </summary> <video src='https://github.com/Expensify/App/assets/95647348/9b7d0277-d199-4fb4-8dcc-1a8bc5f029c0'> </details> | <details> <summary>  ❌ </summary> <video src='https://github.com/Expensify/App/assets/95647348/01fe3e77-8509-4644-a761-67f359a71b72'> </details> |
| - Request money  | <details> <summary>  ❌ </summary> <video src='https://github.com/Expensify/App/assets/95647348/31322b5e-cf7e-49d0-be5f-0ec9655bd42e'> </details> | <details> <summary>  ❌ </summary> <video src='https://github.com/Expensify/App/assets/95647348/f9d3b260-b140-4a87-b85b-a8c247199d08'> </details> |
| - Send money  | <details> <summary>  ❌ </summary> <video src='https://github.com/Expensify/App/assets/95647348/23a2cff6-7371-4171-961f-357223c8c2f8'> </details> | <details> <summary>  ❌ </summary> <video src='https://github.com/Expensify/App/assets/95647348/f637dfc5-d645-4d23-97de-cf322d3b2ab9'> </details> |
| - Assign task  | <details> <summary>  ❌ </summary><video src='https://github.com/Expensify/App/assets/95647348/220737d1-808c-44f6-9750-e5a8c70d86fc'>  </details> | <details> <summary> ❌ </summary> <video src='https://github.com/Expensify/App/assets/95647348/e374873d-6e44-4170-b086-9a166a1fb3a2'> </details> |
| <details> <summary> Assign task - share somewhere </summary> - Press "Assign task" </br> - Fill the title and press "Next" </br> - Press "Share somewhere" </br> </details>  | <details> <summary>  ❌ </summary> <video src='https://github.com/Expensify/App/assets/95647348/a3824939-2660-469e-b970-6ffd91122422'> </details> | <details> <summary> ✅  </summary> <video src='https://github.com/Expensify/App/assets/95647348/15ac7c0b-8db3-474e-8897-f918b9e51bba'> </details> |
| <details> <summary> Invite to workspace </summary> - Open "Choose a workspace" </br> - Press "+" button </br> - Go to "Members" </br> - Press "Invite member" </br> </details>  | <details> <summary> ❌ </summary> <video src='https://github.com/Expensify/App/assets/95647348/9cb89507-f7fc-4302-983e-224b555ea939'> </details> | <details> <summary>  ❌ </summary> <video src='https://github.com/Expensify/App/assets/95647348/aea7b0f7-7cc9-401f-86f0-8634439e22b7'> </details> |
| <details> <summary> Share logs </summary> - Go to Settings -> About -> Troubleshoot </br> - Enable "Client Side logging" </br> - Press "Debug console" </br> - Press "Share log" </br> </details>  | <details> <summary>  ❌   </summary> <video src='https://github.com/Expensify/App/assets/95647348/62d37c14-b7ae-4420-9cbc-e779a3666e34'> </details> | <details> <summary>  ❌ </summary> <video src='https://github.com/Expensify/App/assets/95647348/7a4b1a21-de1b-4747-92a6-a7d9fc1306de'> </details> | 

1. I think the **Skeleton** issue still requires hold. Can we fix them all together in a separate issue/PR later? Since it has different root causes (`didScreenTransitionEnd` + [_"loading searchable lists by caching the options"_](https://github.com/Expensify/App/pull/38207)) from this issue.

2. For the **Green Circular** loading (this issue), I can updat my [PR](https://github.com/Expensify/App/pull/37842) to include all cases. `NewChatPage` still uses `OptionsSelector` we could wait [`Replace OptionsSelector with SelectionList`](https://github.com/Expensify/App/pull/38994) to move on, but I think we are good here, since the green circular on the `NewChatPage` is working now and should keep working after refactor.

@mollfpr @lakchote @mountiny @dukenv0307 Thoughts?

**Comment by mountiny** [comment]
I am up for unifying this

**Comment by brunovjk** [comment]
Perfecto

**Comment by mountiny** [comment]
Unifying means as you wrote:

> For the Green Circular loading (this issue), I can update my https://github.com/Expensify/App/pull/37842 to include all cases.

Just making sure the green loading works same everywhere we would expect this

**Comment by brunovjk** [comment]
There are several places in our App where we use a `SelectionList` with a Search Field, but they **do not** load a **contact list**. 

- src/components/CategoryPicker.tsx
- src/components/DatePicker/CalendarPicker/YearPickerModal.tsx
- src/components/StatePicker/StateSelectorModal.tsx
- src/components/TagPicker/index.tsx
- src/components/TaxPicker.tsx
- src/pages/EditReportFieldDropdown.tsx
- src/pages/EditReportFieldDropdownPage.tsx
- src/pages/iou/IOUCurrencySelection.js
- src/pages/iou/request/step/IOURequestStepCurrency.js
- src/pages/settings/Profile/PersonalDetails/CountrySelectionPage.tsx
- src/pages/settings/Profile/PersonalDetails/StateSelectionPage.tsx
- src/pages/settings/Profile/PronounsPage.tsx
- src/pages/workspace/workflows/WorkspaceAutoReportingMonthlyOffsetPage.tsx
- src/pages/workspace/workflows/WorkspaceWorkflowsApproverPage.tsx or UU?
- src/pages/workspace/workflows/WorkspaceWorkflowsPayerPage.tsx or UU?
- src/pages/workspace/WorkspaceProfileCurrencyPage.tsx
- src/pages/WorkspaceSwitcherPage.tsx
- src/pages/WorkspaceSwitcherPage.tsx

Should we configure the "Green Circular Loading" on them too? I believe the answer is no, but I would like to confirm. Thanks.

**Comment by brunovjk** [comment]
We must also enable the **Green Circular** in [`Room Members Page`](https://github.com/Expensify/App/blob/main/src/pages/RoomMembersPage.tsx)? Thank you.

**Comment by brunovjk** [comment]
[PR](https://github.com/Expensify/App/pull/39865) ready for review

**Comment by brunovjk** [comment]
PR was merged, but not to staging yet

**Comment by lakchote** [comment]
> Payment Summary
> 
> * C: @brunovjk $500
> * C: @dukenv0307 $500
> * C+: @mollfpr $500
> 
> Is this correct? I'll need to create a new Upworks job.

This is correct for @mollfpr and @brunovjk! However not for @dukenv0307 as his PR for handling skeleton cases isn't ready, so no payment is due yet.

**Comment by jliexpensify** [comment]
@lakchote, thanks for checking! I assume I will be paying @dukenv0307 via this job, so we'll need to keep this open?

**Comment by jliexpensify** [comment]
[NEW UPWORKS JOB](https://www.upwork.com/jobs/~01954f725c5457713c)

**Comment by jliexpensify** [comment]
Paid @brunovjk

**Comment by JmillsExpensify** [comment]
$500 approved for @mollfpr

**Comment by jliexpensify** [comment]
Note for myself: waiting to pay @dukenv0307

**Comment by brunovjk** [comment]
Hey @lakchote, @mollfpr, and @dukenv0307,

It's been a couple of weeks since our **fix for the green circular loading** was pushed to production, and it seems to be working fine. Now, I tjhink it's time to **standardize how we handle the skeleton loading on those same pages**.

Most of the pages have the skeleton working fine:

- [x] - src/pages/ChatFinderPage/index.tsx
- [x] - src/pages/NewChatPage.tsx
- [x] - src/pages/InviteReportParticipantsPage.tsx
- [ ] - src/pages/RoomInvitePage.tsx
- [x] - src/pages/iou/request/MoneyRequestParticipantsSelector.ts
    - Submit Expense
    - Split Expense
    - Pay Someone
    - Send Invoice
- [x] - src/pages/settings/AboutPage/ShareLogList/BaseShareLogList.tsx
- [ ] - src/pages/tasks/TaskAssigneeSelectorModal.tsx
- [x] - src/pages/tasks/TaskShareDestinationSelectorModal.tsx
- [x] - src/pages/workspace/WorkspaceInvitePage.tsx

However, a couple of them need proper implementation (`showLoadingPlaceholder={!areOptionsInitialized || !didScreenTransitionEnd}`).
For example, `TaskAssigneeSelectorModal` still uses its own `useOptions`, and not the `useOptionsList` hook like the others. 

@dukenv0307, do you want to work on this, or can I?

Thank you!

**Comment by jliexpensify** [comment]
@lakchote feel free to bump me when @dukenv0307 needs payment.
